crates/core/tedge_agent/src/agent.rs

```diff
@@ -247,7 +247,13 @@ impl SmAgent {
         create_directory_with_user_group(&format!("{config_dir}/.agent"), "tedge", "tedge", 0o775)?;
         create_directory_with_user_group(self.config.log_dir.clone(), "tedge", "tedge", 0o775)?;
         create_directory_with_user_group(
-            &self.config.http_config.file_transfer_dir_as_string(),
+            format!("{DEFAULT_VAR_PATH}/tedge"),
+            "tedge",
+            "tedge",
+            0o775,
+        )?;
+        create_directory_with_user_group(
+            format!("{DEFAULT_VAR_PATH}{HTTP_DEFAULT_URI}"),
```

Comment:
No. This works only if the `HTTP_DEFAULT_URI` starts with `tedge` and has only 2 levels. I know that this unlikely to change but this is fragile and defeats the purpose of using constants.

Actually, the former code is from me and the omission of the `file-transfer` suffix in both the http root and uri was intentional, precisely to avoid this issue.

Reply:
The suffix `file-transfer` was added to the URL in the spec to better namespace this service and avoid any conflicts in future. I'm expecting more HTTP services to be exposed by the tedge-agent like `/tedge/inventory` in the near future.

Reply:
> The suffix file-transfer was added to the URL in the spec to better namespace this service and avoid any conflicts in future. I'm expecting more HTTP services to be exposed by the tedge-agent like /tedge/inventory in the near future. But, if that's causing a lot of impl headache, it can be omitted now and we can revisit this later.

The suffix `file-transfer` makes sense, exactly for the reasons you highlight. I'm not asking to remove it. The `file_transfer_end_point` must be `format!("{}file-transfer/*", self.file_transfer_uri`.